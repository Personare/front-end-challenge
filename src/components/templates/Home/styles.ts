import styled from 'styled-components';
import { colors } from 'styles/variables';

export const Page = styled.div`
  display: grid;
  grid-template-columns: [full-start] 24px [main-start] auto [main-end] 24px [full-end];

  @media screen and (min-width: 768px) {
    grid-template-columns: [full-start] 32px [main-start] auto [main-end] 32px [full-end];
  }

  @media screen and (min-width: 1024px) {
    grid-template-columns: [full-start] 1fr [main-start] 755px [main-end] 1fr [full-end];
  }
`;

export const Main = styled.main`
  grid-column: main;
  padding-top: 48px;
`;

export const Title = styled.h1`
  color: ${colors.blue};
  font-size: 42px;
  font-weight: 700;
  text-transform: uppercase;
`;

export const Box = styled.div`
  background: ${colors.white};
  box-shadow: 2px 2px 8px ${colors.gray};
  display: flex;
  justify-content: center;
  margin-top: 48px;
  padding: 24px 16px;
  width: 100%;

  @media screen and (min-width: 768px) {
    padding: 32px 48px;
  }

  @media screen and (min-width: 1024px) {
    padding: 32px 24px;
  }
`;

export const Text = styled.span`
  color: ${colors.blue};
  display: block;
  font-size: 20px;
  letter-spacing: 0.05em;
  line-height: 130%;
  text-align: center;
  text-transform: uppercase;
`;
