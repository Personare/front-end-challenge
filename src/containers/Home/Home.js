import React, { Component } from 'react'
import styled from 'styled-components'
import { Button, Card, Modal } from '../../components'
import getCards from '../../services'
import { randomicArray } from '../../utils/utils'
import { CARD_DESCRIPTION } from '../../utils/messages'

export default class Home extends Component {
  constructor() {
    super()

    this.state = {
      cards: [],
      imagesUrl: '',
      imageBackCard: '',
      hasStart: false,
      showModalFlag: false,
      currentCard: {
        name: '',
        image: ''
      }
    }

    this.startGame = this.startGame.bind(this)
    this.choosenCard = this.choosenCard.bind(this)
    this.closeModal = this.closeModal.bind(this)
    this.restartGame = this.restartGame.bind(this)
  }

  componentDidMount() {
    getCards().then(response => {
      this.setState({
        cards: response.cards,
        imagesUrl: response.imagesUrl,
        imageBackCard: response.imageBackCard
      })
    })
  }

  startGame() {
    const { cards } = this.state
    const randomicCards = randomicArray(cards)
    this.setState({ hasStart: true, cards: randomicCards })
  }

  restartGame() {
    const { cards } = this.state
    this.closeModal()
    this.setState({ hasStart: false, cards })
  }

  choosenCard(name, image) {
    const { hasStart } = this.state
    if (hasStart) {
      this.setState({
        showModalFlag: true,
        currentCard: {
          name,
          image
        }
      })
    }
  }

  closeModal() {
    this.setState({ showModalFlag: false })
  }

  renderCardList() {
    const {
      cards, imagesUrl, imageBackCard, hasStart
    } = this.state

    return cards.map(({ name, image }) => (
      <Card
        key={name}
        image={image}
        imagesUrl={imagesUrl}
        imageBackCard={imageBackCard}
        hasStart={hasStart}
        onClick={() => this.choosenCard(name, image)}
      />
    ))
  }

  render() {
    const {
      imagesUrl, imageBackCard, showModalFlag, currentCard, hasStart
    } = this.state
    const { name, image } = currentCard

    return (
      <HomeContainer>
        {!hasStart && <Button onClick={this.startGame}>Start</Button>}

        {showModalFlag && (
          <Modal onClose={this.closeModal} title="Carta escolhida">
            <Card
              key={name}
              description={CARD_DESCRIPTION}
              name={name}
              image={image}
              imagesUrl={imagesUrl}
              imageBackCard={imageBackCard}
              width="200px"
              height="400px"
              cardBoard={false}
            />
            <Button onClick={this.restartGame}>Play Again</Button>
          </Modal>
        )}
        <CardsContainer>{this.renderCardList()}</CardsContainer>
      </HomeContainer>
    )
  }
}

const HomeContainer = styled.div``

const CardsContainer = styled.div`
  display: flex;
  flex-wrap: wrap;
`
