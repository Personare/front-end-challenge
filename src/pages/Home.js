import {useEffect, useState, useContext} from 'react'
import TarotCard from '../components/TarotCard'
import styled from 'styled-components'
import axios from 'axios'
import Context from '../global/Context'


const Container = styled.div`
	display: grid;
	grid-template-columns: repeat(4, 1fr);
`

//====================Componente funcional===========
const Home = (props)=>{
	const {states, setters, requests} = useContext(Context)


	useEffect(()=>{
		requests.getCards()
	}, [])

		
	
	return<div><p className='btn'><button onClick={()=> props.changePage('cheap')}>Jogar</button></p>
			<Container>			
				{states.cards.map(card=>{
					return<TarotCard key={card.name}
							name={card.name}
							image={card.image}/>
				})}
		    </Container> 
			  
		  </div>
}
export default Home
