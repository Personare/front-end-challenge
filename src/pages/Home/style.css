.cardShowed{
	display: grid;
	grid-template-columns: repeat(5, 1fr);
	text-align: center;
}

.btn{
	text-align: center;
	margin: 30px;
}

.btn-play{
	width: 150px;
	height: 35px;
	font-size: 1.5rem;
	background-color: blue;
	border: none;
	border-radius: 10px;
	color: whitesmoke;
	cursor: pointer;
}

.cards{
	margin: 10px;
}