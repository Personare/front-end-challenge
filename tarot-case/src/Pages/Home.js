
import { Body } from "../Styled/styledPages"
import { goToTable } from "../Router/cordination"
import { useHistory } from "react-router-dom"
import Button from '@material-ui/core/Button';

const Home = () => {

    const history = useHistory();

    const jogar = () => {
        goToTable(history)
    }

    return (
        <Body>

            <h1> BEM VIND@!</h1>
            <h3>Está preparad@ para ver o seu futuro?</h3>
            <p>Como Funciona?</p>
            <li>Para começar, clique em embaralhar</li>
            <li>Escolha uma carta e clique nela para ver o que ela lhe diz</li>
            <li>Para voltar ao tabuleiro principal basta clicar novamente em cima da carta que apareceu.</li>
          <p> obs: As cartas são embaralhadas novamente sempre que você volta ao tabuleiro principal</p>
            <hr/>
            <h3> Vamos lá!</h3>
            <Button onClick={jogar}
                variant="contained" color="primary">START</Button>
        </Body>

    )
}
export default Home