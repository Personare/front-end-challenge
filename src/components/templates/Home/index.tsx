import React, { MouseEvent, useState } from 'react';

import Button from 'components/atoms/Button';
import Cards, { CardsProps } from 'components/organisms/Cards';

import * as S from './styles';

const Home = (props: CardsProps) => {
  const [activated, setActivated] = useState(false);

  function handleClickButton(e: MouseEvent<HTMLButtonElement>) {
    setActivated(true);
    e.currentTarget.disabled = true;
  }

  return (
    <S.Page>
      <S.Main>
        <S.Title>Tarot</S.Title>

        <S.Box>
          <Button text="Iniciar o jogo" onClick={handleClickButton} />
        </S.Box>

        <S.Box>
          <S.Text>Selecione uma carta do baralho abaixo</S.Text>

          <S.CardsWrapper>
            <Cards activated={activated} data={{ ...props }} />
          </S.CardsWrapper>
        </S.Box>
      </S.Main>
    </S.Page>
  );
};

export default Home;
