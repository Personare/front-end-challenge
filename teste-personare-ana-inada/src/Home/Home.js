import React, { useEffect, useState } from "react";
import { styled } from "@mui/material/styles";
import Grid from "@mui/material/Grid";
import Paper from "@mui/material/Paper";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import Modal from "@mui/material/Modal";
import { CARDS } from "../cards";
const Item = styled(Paper)(({ theme }) => ({
    backgroundColor: theme.palette.mode === "dark" ? "#1A2027" : "#fff",
    ...theme.typography.body2,
    padding: theme.spacing(1),
    textAlign: "center",
    color: theme.palette.text.secondary,
}));

// Função para randomizar array
function embaralhar(cards) {
    // Loop em todos os elementos
    for (let i = cards.length - 1; i > 0; i--) {
        // Escolhendo elemento aleatório
        const j = Math.floor(Math.random() * (i + 1));
        // Reposicionando elemento
        [cards[i], cards[j]] = [cards[j], cards[i]];
    }
    // Retornando array com aleatoriedade
    return cards;
}

export const Home = () => {
    const [cards, setCards] = useState(CARDS.cards);
    const [newCard, setNewCard] = useState();
    const changeCard = (card) => {
        setNewCard(card);
        embaralhar(cards);
    };
    const handleClose = () => {
        setNewCard();
    };

    useEffect(() => {
        setCards(embaralhar(cards));
    }, []);
    return (
        <Box sx={{ width: "100%" }}>
            <Grid
                container
                rowSpacing={1}
                columnSpacing={{ xs: 1, sm: 2, md: 3 }}
                columns={{ xs: 1, sm: 4, md: 8, lg: 13 }}
            >
                {cards.map((card) => (
                    <Grid item key={card.name} xs={1}>
                        <Item>
                            <img
                                onClick={() => changeCard(card)}
                                src={CARDS.imageBackCard}
                            ></img>
                        </Item>
                    </Grid>
                ))}
            </Grid>
            <Modal
                open={!!newCard}
                onClose={handleClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <Box
                    sx={{
                        p: 6,
                        backgroundColor: "white",
                        width: 0.8,
                        margin: "auto",
                    }}
                >
                    <Typography
                        id="modal-modal-title"
                        variant="h6"
                        component="h2"
                    >
                        {newCard?.name}
                    </Typography>
                    <Typography id="modal-modal-description" sx={{ mt: 2 }}>
                        Duis mollis, est non commodo luctus, nisi erat porttitor
                        ligula.
                    </Typography>
                    <img src={`${CARDS.imagesUrl}${newCard?.image}`}></img>
                </Box>
            </Modal>
        </Box>
    );
};

export default Home;
