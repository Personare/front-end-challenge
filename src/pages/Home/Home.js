import {useState, useEffect} from 'react'
import axios from 'axios'
import './style.css'


const Home = (props)=>{
	const [cards, setCards] = useState([])

	useEffect(()=>{
    axios.get('tarot.json').then(res=>{
      setCards(res.data.cards)
    }).catch(err=>{
      alert('Algo deu errado!\n'+err.response)
    })
  }, [])



	const imageUrl = 'https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/images/content/pt-br/product/tarot/marselha/162x341'	

	return<div>			
			<div className='btn'>
			<button className='btn-play' onClick={()=> props.changePage('intro')}>Jogar</button>
			</div>
			<div className='cardShowed'>
			{cards.length > 0 ? cards.map(card=>{
				return <div className='cards' key={card.name}>
						{card.name}
						<div>
							<img className='character' src={`${imageUrl}/${card.image}`}/>
						</div>
					   </div>
			}) : <div className='load-container'>
					<div className='loading'></div>
				</div>}
		  </div>
		</div>
}
export default Home