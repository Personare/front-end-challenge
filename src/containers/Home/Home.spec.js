import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import Home from './Home'

describe('Home', () => {
  const state = {
    cards: [
      {
        image: 'arcano1.jpg',
        name: 'O MAGO'
      },
      {
        image: 'arcano2.jpg',
        name: 'A SACERDOTISA'
      }
    ],
    currentCard: {
      image: 'arcano2.jpg',
      name: 'A SACERDOTISA'
    },
    hasStart: false,
    imageBackCard: `https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/
      images/content/pt-br/product/tarot/marselha/back-blue-card.png`,
    imagesUrl: `https://dkw5ssdvaqf8l.cloudfront.net/static/psr/br/framework/yii/
      content/pt-br/product/tarot/marselha/162x341/`,
    showModalFlag: false
  }

  it('should render correctly', () => {
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should startGame when clicks on start button', () => {
    const tree = mount(<Home />)
    tree.find('button').simulate('click')
    expect(tree.state().hasStart).toBe(true)
  })

  it('should show modal when clicks on card', () => {
    const image = 'arcano1.jpg'
    const name = 'O MAGO'
    const tree = mount(<Home />)
    tree.setState({ ...state, hasStart: true })

    tree
      .find('Home')
      .first()
      .instance()
      .choosenCard(name, image)
    expect(tree.state().showModalFlag).toBe(true)
  })

  it('should close modal when call closeModal function', () => {
    const tree = mount(<Home />)
    tree.setState({ ...state, hasStart: true, showModalFlag: true })

    tree
      .find('Home')
      .first()
      .instance()
      .closeModal()

    expect(tree.state().showModalFlag).toBe(false)
  })

  it('should restart game when restartGame is called', () => {
    const tree = mount(<Home />)
    tree.setState({ ...state, hasStart: true, showModalFlag: true })

    tree
      .find('Home')
      .first()
      .instance()
      .restartGame()

    expect(tree.state()).toEqual(state)
  })
})
