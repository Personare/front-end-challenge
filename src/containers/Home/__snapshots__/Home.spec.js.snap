// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render correctly 1`] = `
<div
  className="sc-gqjmRU hnZXkc"
>
  <button
    className="sc-bdVaJa duNfdq"
    onClick={[Function]}
    type="button"
  >
    Start
  </button>
  <div
    className="sc-VigVT UDKFG"
  />
</div>
`;
